src/Voyager.php

```diff
@@ -68,12 +68,24 @@ public static function can($permission)
 
         if ($exist) {
             $user = User::find(Auth::id());
-            if ($user == null) {
+            if ($user == null || !$user->hasPermission($permission)) {
                 throw new UnauthorizedHttpException(null);
             }
-            if (!$user->hasPermission($permission)) {
-                throw new UnauthorizedHttpException(null);
+        }
+    }
+
+    public static function have($permission)
+    {
+        // Check if permission exist
+        $exist = Permission::where('key', $permission)->first();
+
+        if ($exist) {
+            $user = User::find(Auth::id());
```

Comment:
Maybe the `User` model should check if the relationship is loaded and else load it instead. Then it will also be working for other methods than `Voyager::for`.